data/conf/iceberg/llap/hive-site.xml

```diff
@@ -38,10 +38,12 @@
         <description>A base for other temporary directories.</description>
     </property>
 
+    <!-- in LLAP mode hive.tez.container.size isn't used to actually determine container size, however while -->
```

Comment:
are we estimating not optimal number of slots in case of LLAP? We are relying on `hive.tez.container.size` that's not used in LLAP sizing?

Reply:
good question, so in case of a testing llap driver (e.g. TestMiniLlapLocalCliDriver), where there is no classic llap deamons at all, this was what I saw...however, you're pointing out a new confusion, yeah, so if hive.tez.container.size is not inline with the independently sized LLAP deamons, then this is a discrepancy in available slot calculation...let me check this for a LLAP product, most probably Cloudera's CDW, and bring clarity here, at least in comments

Reply:
Intresting discussion. I want to learn something about llap too. :)
IMHO, `hive.tez.container.size` has no effect LLAP task(exector) memory size. It seems that llap task size is decided by this code:
https://github.com/apache/hive/blob/36ce858163a19e29eafe4a8d3307191bc28fc175/llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/ContainerRunnerImpl.java#L164-L165

Reply:
yeah, hive.tez.container.size has no effect on the actual executor memory size (that's handled on LLAP codepath, e,g, what you linked, part of llap deamon sizing), but it seems to have an effect on available slots calculation, which is a crucial part of the split generation, that's why I'll have to clarify this area in code comments

Reply:
Thanks for explanation. Maybe i am not familar about llap implementation. But it seems strange that llap split generation is related `hive.tez.container.size` instead of per task(executor) size.
Maybe i needI to take some time to learn this part. :)

Reply:
I know how it feels :) I hope by the end of this PR we'll all have a better understanding of this...

Reply:
@zhangbutao , @deniskuzZ : please find the latest comment and let me know if it's clearer

Reply:
Thx. I think this change is clear enough to me.

Reply:
thanks @zhangbutao ! 
@zabetak , @deniskuzZ  : please let me know if the current state of this PR is fine enough to be merged (it's important, I wanted to bring clarity, so please confirm if I succeeded)